### ISSUE TITLE
Error in cp1148

### ISSUE THREAD

**Issue by fdegimbe** [roskakori/CodecMapper#4]
I believe there is an error in the cp1148 codec.

'\x85'.encode('cp1148')
expected result :  b'\x15'
raises: UnicodeEncodeError: 'charmap' codec can't encode character '\x85' in position 0: character maps to <undefined>

b'\x15'.decode('cp1148')
expected result:  '\x85'
returns:  '\n'

You can test with cp500. The only difference being that b'\x9f' will be decoded to the currency sign ('¤') istead of the euro sign (€).

**Comment by roskakori** [comment]
The reason for the difference between cp500 and cp1141 is that cp500 is provided by the Python standard library while cp1148 is part of the ebcdic package.

The cp1148 mapping is generated from the cp1148 mapping included with the Java standard library. indeed 0x85 is not mapped there.

According to the [Wikipedia page on cp500](https://en.wikipedia.org/wiki/EBCDIC_500), 0x85 should be mapped to 0x15 as you described. However referred PDF about cp500 does not even include the range 0x00 to 0x3f. It's from IBM from 1986 so it probably is pretty outdated by now.

AFAIR cp500 has been a moving target for quite a while (as has been cp1552, the "pimped" ISO-8859-1).

I agree this should be fixed so that cp1141 is "cp500 with Euro sign".

**Comment by roskakori** [comment]
I digged some more and the cp500 included with the Python 3.6.8 is generated from a file named `MAPPINGS/VENDORS/MICSFT/EBCDIC/CP500.TXT`. So apparently this is a mapping from Microsoft instead of IBM and naturally they include their extensions for ISO-8859-1 that are also part of cp1252 (AKA "Windows ANSI"). This also seems to be the mapping used (but not referenced) by Wikipedia.

Java however was long a an important asset for IBM and it seems to still use a cp500 that maps only characters from ISO-8859-1. Which technically speaking is correct.

Regardless, in Python the encodings of the ebcdic package that are just Euro variants of encodings in the standard library should be consistent.

**Comment by roskakori** [comment]
Oh well, this is a mess:

Microsoft provides mappings for their interpretation of EBCDIC at <https://www.unicode.org/Public/MAPPINGS/VENDORS/MICSFT/EBCDIC/>. However these are only a few mappings including cp500 but not cp1148.

IBM on the other hand provides mappings for both cp500 andcp1148, which are the same only for the Euro sign. However the IBM cp500 differs from the Microsoft cp500.

**Comment by roskakori** [comment]
@fdegimbe I solved this by adding cp500ms and cp1148ms which are the Microsoft interpretations. I compared Python's cp500 with Java's cp500 and the only difference is the mapping of 0x1f to 0x85 ("next line") respectively 0x0a ("linefeed").

So if you want a cp1148 that differs from the Python standard libraries cp500 only by the Euro sign you'll have to use cp1148ms instead if cp1148 (which includes the IBM interpretation).

The improved version is already available from https://pypi.org/project/ebcdic/
